flict/flictlib/translator.py

```diff
@@ -94,61 +98,68 @@ def update_license(translations, license_expr):
 def update_packages(translations, dependencies):
     updates_deps = []
     for dep in dependencies:
-        #        print("license: \"" + dep["license"] + "\"")
         license = dep["license"].strip()
         dep["license"] = update_license(translations, license)
         dep_deps = dep["dependencies"]
         updates_deps = update_packages(translations, dep_deps)
     return updates_deps
 
 
-def _read_translations(translations_file):
-    with open(translations_file) as fp:
-        translations_object = json.load(fp)
-        translations = translations_object["translations"]
-
-        translations_map = {}
-        for item in translations:
-            if item['translation'] is not None:
-                translations_map[item['value']] = item
-
-        return {
-            'original': translations_object,
-            'translations_map': translations_map
-        }
-
-
 def read_translations(translations_file):
     symbols = {}
-    with open(translations_file) as fp:
-        translations_object = json.load(fp)
-        translations = translations_object["translations"]
-        for item in translations:
-            #print(" let's check...." + str(item), end="")
-            if 'license_expression' in item and 'spdx_id' in item:
-                le = item['license_expression'].lower()
-                if "scancode" in le:
-                    logger.main_logger.debug(" IGNORING since scancode")
-                    pass
-                else:
-                    #print(" OK ")
-                    transl = item['spdx_id']
-                    key = item['license_expression']
-                    if transl not in symbols:
-                        symbols[transl] = []
-                    symbols[transl].append(key)
+    try:
+        with open(translations_file) as file_:
+            translations_object = json.load(file_)
+    except (FileNotFoundError, json.JSONDecodeError):
+        raise RuntimeError("reading translations file failed")
```

Comment:
We are using our own Exception set for all kinds of exception to encountered - please have a look at https://github.com/vinland-technology/flict/blob/e2a15bdcc59d3de72558d817a26e6f4ef5344a94/flict/flictlib/return_codes.py#L50 and turn it into a FlictError with appropriate error code

Reply:
indeed, i have seen this in other files. However, this would add another dependency to be resolved during importing symbol. Perhaps this file was never meant to be a stand-alone script, which could be used separately?
I could use RET_FILE_NOT_FOUND and RET_INVALID_EXPRESSSION - but i think it was purposely just for invalid license expression. Should one create a new return code, for general errors in input JSON file?